plugins/wordcompletion/pluma-word-completion-plugin.c

```diff
@@ -0,0 +1,370 @@
+                                G_IMPLEMENT_INTERFACE_DYNAMIC (PEAS_GTK_TYPE_CONFIGURABLE,
+                                                               peas_gtk_configurable_iface_init)
+                                G_ADD_PRIVATE_DYNAMIC (PlumaWordCompletionPlugin))
+
+enum
+{
+    PROP_0,
+    PROP_WINDOW,
+    PROP_VIEW
+};
+
+static void
+pluma_word_completion_plugin_init (PlumaWordCompletionPlugin *plugin)
+{
+    pluma_debug_message (DEBUG_PLUGINS, "PlumaWordCompletionPlugin initializing");
+
+    plugin->priv = pluma_word_completion_plugin_get_instance_private (plugin);
+}
+
+static void
+pluma_word_completion_plugin_dispose (GObject *object)
+{
+    PlumaWordCompletionPlugin *plugin = PLUMA_WORD_COMPLETION_PLUGIN (object);
+
+    g_clear_object(&plugin->priv->window);
+    g_clear_object(&plugin->priv->view);
+    g_clear_object(&plugin->priv->provider);
+
+    G_OBJECT_CLASS (pluma_word_completion_plugin_parent_class)->dispose (object);
+}
+
+static void
+pluma_word_completion_plugin_set_property (GObject      *object,
+                                           guint         prop_id,
+                                           const GValue *value,
+                                           GParamSpec   *pspec)
+{
+    PlumaWordCompletionPlugin *plugin = PLUMA_WORD_COMPLETION_PLUGIN (object);
+
+    switch (prop_id)
+    {
+        case PROP_WINDOW:
+            plugin->priv->window = g_value_dup_object (value);
+            break;
+        case PROP_VIEW:
+            plugin->priv->view = PLUMA_VIEW (g_value_dup_object (value));
+            break;
+        default:
+            G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+            break;
+    }
+}
+
+static void
+pluma_word_completion_plugin_get_property (GObject    *object,
+                                           guint       prop_id,
+                                           GValue     *value,
+                                           GParamSpec *pspec)
+{
+    PlumaWordCompletionPlugin *plugin = PLUMA_WORD_COMPLETION_PLUGIN (object);
+
+    switch (prop_id)
+    {
+        case PROP_WINDOW:
+            g_value_set_object (value, plugin->priv->window);
+            break;
+        case PROP_VIEW:
+            g_value_set_object (value, plugin->priv->view);
+            break;
+        default:
+            G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+            break;
+    }
+}
+
+static void
+update_activation (GtkSourceCompletionWords *provider,
+                   GSettings                *settings)
+{
+    GtkSourceCompletionActivation activation;
+
+    g_object_get (provider, "activation", &activation, NULL);
+
+    if (g_settings_get_boolean (settings, SETTINGS_KEY_INTERACTIVE_COMPLETION))
+    {
+        activation |= GTK_SOURCE_COMPLETION_ACTIVATION_INTERACTIVE;
+    }
+    else
+    {
+        activation &= ~GTK_SOURCE_COMPLETION_ACTIVATION_INTERACTIVE;
```

Comment:
```
pluma-word-completion-plugin.c:146:20: warning: unsigned conversion from 'int' to 'unsigned int' changes value from '-2' to '4294967294' [-Wsign-conversion]
  146 |         activation &= ~GTK_SOURCE_COMPLETION_ACTIVATION_INTERACTIVE;
      |                    ^~
```

Reply:
This is because of this commit (https://gitlab.gnome.org/GNOME/gtksourceview/-/blob/3644449c16935cbb5f3afd299f5764166832155f/gtksourceview/gtksourcecompletioncontext.h#L35) which changed `1 << 1` to `2`, hence an unsigned 2 to a signed one.
simple fix: `(~(unsigned int)GTK_SOURCE_COMPLETION_ACTIVATION_INTERACTIVE)`